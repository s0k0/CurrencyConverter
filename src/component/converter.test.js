
import React from 'react';
import axios from 'axios';
import { render, waitForElement } from '@testing-library/react';
import Converter from './converter';

jest.mock('axios')

function mockCall() {
  axios.get.mockResolvedValueOnce({
    data: {
       base: 'GBP',
       rates: { EUR: 1.1234}
    }
  });
} 

describe('Converter component', () => {

  test('renders converter component', () => {
    expect(Converter).toMatchSnapshot()
  });

  test('should fetch rates', async() => {
    mockCall();
    const url = 'https://cors-anywhere.herokuapp.com/http://api.openrates.io/latest?base=GBP&symbols=EUR'
    const { getByText } = render(<Converter />);
    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith(url);
    //check whether exchange rate from mock response is received and rendered into component
    const rate = await waitForElement(() => getByText(/1.12/i))
    expect(rate).toBeInTheDocument();
  });
});
