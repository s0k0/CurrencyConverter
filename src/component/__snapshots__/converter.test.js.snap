// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter component renders converter component 1`] = `[Function]`;
